File: examples/hello/hello.rb
require 'json'

name = "World"

payload = STDIN.read
if payload != ""
    puts "Received from STDIN: #{payload}"
end

# or using env vars: ENV['PAYLOAD']
payload = ENV['PAYLOAD']

STDERR.puts 'payload: ' + payload.inspect
if payload != ""
    payload = JSON.parse(payload)
    name = payload['name']
end

puts "Hello #{name}!"

File: examples/hello/python3/hello.py
import sys
sys.path.append("packages")
import os
import json

name = "World"
if not os.isatty(sys.stdin.fileno()):
    obj = json.loads(sys.stdin.read())
    if obj["name"] != "":
        name = obj["name"]

print("Hello ", name, "!!!",
      " from python version",
      str(sys.version_info.major) + '.' +
      str(sys.version_info.minor) + '.' +
      str(sys.version_info.micro))
